src/rkopenmdao/error_controller.py

```diff
@@ -0,0 +1,74 @@
+from dataclasses import dataclass
+
+from rkopenmdao.error_estimator import ErrorEstimator
+
+
+def _three_queue_controller(_list: list, val) -> list:
+    if _list is None:
+        _list = [val]
+    elif len(_list) < 3:
+        _list.append(val)
+    else:
+        _list.pop(0)
+        _list.append(val)
+    return _list
+
+
+@dataclass
+class ErrorController:
+    alpha: float
+    beta: float = 0
+    gamma: float = 0
+    a: float = 0
+    b: float = 0
+    name: str = "ErrorController"
+    local_error_norms: list = None
+    delta_time_steps: list = None
+
+    def _estimate_next_step_function(self,
+                                     tolerance: float = 1e-3,
+                                     safety_factor: float = 0.95):
+
+        delta_time_new = safety_factor * self.delta_time_steps[-1]
+        if 0 in self.local_error_norms:
+            return 1.0
```

Comment:
1.0 doesn't seem like the correct return value. I would just return the old step size in that case, though there might be some literature out there that might deal with that edge case.
I highly doubt that that case will happen anyway.

Reply:
changed to        
```
if 0 in self.local_error_norms:
   return self.delta_time_steps[-1]
```